dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/preheat/DefaultTrackerPreheatService.java

```diff
@@ -141,9 +146,13 @@ public TrackerPreheat preheat( TrackerPreheatParams params )
         Timer timer = new SystemTimer().start();
 
         TrackerPreheat preheat = new TrackerPreheat();
+        preheat.setIdentifiers( params.getIdentifiers() );
         preheat.setUser( params.getUser() );
         preheat.setDefaults( manager.getDefaults() );
 
+        // TODO: Morten/Stian could this be done earlier, and rather not allow user to be set to NULL above, ?
+        //  since this has big security implication it would be nicer to separate out this to
+        //  a dedicated security controller/place...?
```

Comment:
I would very much try to avoid using null as a valid value for users. I think it is better to have a clearly defined test user instead. Using null here makes it hard to reason if it is a null pointer bug or intentional. Since at many places a NULL value gives super user access this is even more dangerous.